### PR TITLE
Upstream pull 4e528486931ec523a32f91f564a6a26b3b3b4fce

### PR DESCRIPTION
**Pull request by pellared**
Upstream pull 4e528486931ec523a32f91f564a6a26b3b3b4fce

### CODE REVIEW

File: profiler/build/PiComputation.linux.net50.json [hunk 1]
```diff
@@ -41,9 +42,9 @@
   "environmentVariables": {
     "CORECLR_ENABLE_PROFILING": "1",
     "CORECLR_PROFILER": "{BD1A650D-AC5D-4896-B64F-D6FA25D6B26A}",
-    "CORECLR_PROFILER_PATH_64": "$(CWD)/../../DDProf-Deploy/Datadog.AutoInstrumentation.Profiler.Native.x64.so",
-    "SIGNALFX_DOTNET_PROFILER_HOME": "$(CWD)/../../DDProf-Deploy/",
-    "LD_PRELOAD": "$(CWD)/../../DDProf-Deploy/Datadog.Linux.Wrapper.x64.so",
+    "CORECLR_PROFILER_PATH_64": "$(CWD)/../../shared/bin/monitoring-home/Datadog.Trace.ClrProfiler.Native.so",
```
Comment (RassK) [inline profiler/build/PiComputation.linux.net50.json:45]:
```suggestion
    "CORECLR_PROFILER_PATH_64": "$(CWD)/../../shared/bin/monitoring-home/SignalFx.Tracing.ClrProfiler.Native.so",
```

File: profiler/build/PiComputation.linux.netcoreapp31.json [hunk 2]
```diff
@@ -40,10 +41,10 @@
   "workingDirectory": "$(CWD)/../_build/bin/Release-x64/profiler/src/Demos/Samples.Computer01/netcoreapp3.1",
   "environmentVariables": {
     "CORECLR_ENABLE_PROFILING": "1",
-    "CORECLR_PROFILER": "{BD1A650D-AC5D-4896-B64F-D6FA25D6B26A}",
-    "CORECLR_PROFILER_PATH_64": "$(CWD)/../../DDProf-Deploy/Datadog.AutoInstrumentation.Profiler.Native.x64.so",
-    "SIGNALFX_DOTNET_PROFILER_HOME": "$(CWD)/../../DDProf-Deploy/",
-    "LD_PRELOAD": "$(CWD)/../../DDProf-Deploy/Datadog.Linux.Wrapper.x64.so",
+    "CORECLR_PROFILER": "{846F5F1C-F9AE-4B07-969E-05C26BC060D8}",
+    "CORECLR_PROFILER_PATH_64": "$(CWD)/../../shared/bin/monitoring-home/Datadog.Trace.ClrProfiler.Native.so",
```
Comment (RassK) [inline profiler/build/PiComputation.linux.netcoreapp31.json:45]:
```suggestion
    "CORECLR_PROFILER_PATH_64": "$(CWD)/../../shared/bin/monitoring-home/SignalFx.Tracing.ClrProfiler.Native.so",
```

File: docker-compose.yml [hunk 3]
```diff
@@ -482,6 +482,29 @@ services:
     environment:
       - SIGNALFX_CLR_ENABLE_NGEN=${SIGNALFX_CLR_ENABLE_NGEN:-1}
 
+  ProfilerIntegrationTests:
+    build:
+      context: ./tracer/build/_build/
+      dockerfile: docker/${baseImage:-debian}.dockerfile
+      args:
+        - DOTNETSDK_VERSION=${dotnetCoreSdkLatestVersion:-6.0.100}
```
Comment (Kielek) [inline docker-compose.yml:490]:
```suggestion
        - DOTNETSDK_VERSION=${dotnetCoreSdkLatestVersion:-6.0.301}
```

File: docker-compose.yml [hunk 4]
```diff
@@ -482,6 +482,29 @@ services:
...
+    cap_add:
+      - SYS_PTRACE
+    environment:
+      - NugetPackageDirectory=/project/${relativeNugetPackageDirectory:-packages}
+      - tracerHome=/project/${relativeTracerHome:-shared/bin/monitoring-home/tracer}
+      - profilerHome=/project/${relativeTracerHome:-shared/bin/monitoring-home/continuousprofiler}
+      - artifacts=/project/${relativeArtifacts:-tracer/src/bin/artifacts}
+      - framework=${framework:-netcoreapp3.1}
+      - baseImage=${baseImage:-default}
+      - DD_CLR_ENABLE_NGEN=${DD_CLR_ENABLE_NGEN:-1}
```
Comment (Kielek) [inline docker-compose.yml:504]: DD_ -> SIGNALFX_ - please check whole project

Comment (pellared) [reply docker-compose.yml:504]:
```suggestion
    - SIGNALFX_CLR_ENABLE_NGEN=${SIGNALFX_CLR_ENABLE_NGEN:-1

Comment (pellared) [reply docker-compose.yml]: Fixed https://github.com/signalfx/signalfx-dotnet-tracing/pull/533/commits/2defc813bb380337f29d331b6c1765e86f82769f

**Review by RassK — APPROVED** [review]

**Pull request merged**